### DISCUSSION TITLE
scplot doesn´t connect phases

### DISCUSSION THREAD

**Discussion by lisalizzard** [jazznbass/scan#136] [Q&A] [answered]
Hello:) 
I am trying to create a plot with scplot for a single-case-design with multiple phases. I´d like to use scplot rather than just plot, due to the numerous design options. Unfortunatly scplot doesn´t quite work and I don´t know why. Maybe someone could help?

I coded my phases according to the scan package and when I plot them with plot, there are absolutely no problems. But when I use scplot the graph doesn´t connect the points between the different phases. So for example it generates a mini graph for phase A and phase B, but they are not connected at the "border". 
My phase design is as follows: Phases <-c(A1=14, B1=7, CB1.1=7, B2=7, CB1.2=7, B3=7, A2=7)
I  numbered the phases because otherwise it would connect all A phases, all B phases etc. but not consider the sequence.

I appreciate every help and tip! Thank you in advance :)

**Comment by researchtool** [comment]
Hi Lisa.
Could you provide a minimal code example to reproduce your design? You can just make up values and phases. But to really look into it we'll need an example to reproduce the problem.
Best,
timo

**Reply by lisalizzard** [reply]
Oh of course!
I´ll just give you one dependent variable as example, but i created a dataframe where i collected multiple, but here is stress:

data_stress<-data.frame(efaLS_2$stress1, efaLS_2$stress2, efaLS_2$stress3, efaLS_2$stress4, efaLS_2$stress5)
stress<-rowMeans(data_stress)

AV<-data.frame(efaLS_2$condition, efaLS_2$act, efaLS_2$screen, stress, pos_wb, neg_wb, energy, WB)

Phases <-c(A1=14, B1=7, CB1.1=7, B2=7, CB1.2=7, B3=7, A2=7)

AV_Stress <- scdf(values = stress, phase_design = Phases, name = "ID 9")

install.packages("devtools")
library(devtools)
library(scplot)

scplot(AV_Stress)

I hope that makes sense and helps :)

**Comment by researchtool** [comment, marked as answer]
It helped me understand the structure of your SCs. I made up a reproducible example. Please check (by simply copying and executing) my code and tell me if it differs from your SCs (regarding the structure etc - the values are made up).
```
library(scan)
library(scplot)
stress <- rep(c(4,6,5,6,7,8,7,10),7) #I made up values for this purpose. Please check if the structure corresponds with your data.
Phases <-c(A1=14, B1=7, CB1.1=7, B2=7, CB1.2=7, B3=7, A2=7)
AV_Stress <- scdf(values = stress, phase_design = Phases, name = "ID 9")
AV_Stress #Does this basically represent the structure of your SCs?
scplot(AV_Stress) %>% set_theme("basic") #Does this basically represent your plot?
```
For me the resulting plot looks like the attachment. Is it any different for you? For me scplot does exactly what it should. This is how the plot should look like.
If your plot looks different: Could you use my code and upload the resulting plot?
If your plot looks the same: I probably didn't understand what exactly did not work for you. Maybe you could elaborate given the example.
![Rplot](https://github.com/jazznbass/scan/assets/5195357/2d283755-55c2-4399-beb5-a392637ea244)
